### PR TITLE
Add support for Super Swing Golf Season 2

### PR DESCRIPTION
**Pull request by iwubcode**
This adds support for Super Swing Golf Season 2 (and likely the first game as well).  Going from 0 textures to 11'722!

### CODE REVIEW

**Comment by iwubcode** [comment]
Still palette data is not quite right.  Big thanks to @Venomalia for all the help with it.  Marking as draft for now.

**Comment by Venomalia** [comment]
I think we still need a mipmap flag.
You should be able to find it easily, if you check after adding the texture if the stream is at the end and set a breakpoint.
when it hits, this texture will have mips and you only need to find the flag that is different.

or you use `format.GetMipmapsFromSize()`

File: lib/AuroraLip/Common/FormatInfo.cs [hunk 1]
```diff
@@ -202,7 +202,7 @@ public string GetTypName()
         /// <inheritdoc />
         public virtual bool Equals(FormatInfo other)
         {
-            if (Identifier != null)
+            if (Identifier != null && other.Identifier != null)
             {
                 return Identifier.Equals(other.Identifier);
```
Comment (Venomalia) [inline lib/AuroraLip/Common/FormatInfo.cs:207]: The problem will be that `other` is null, we should check this for null.

`                return Identifier.Equals(other?.Identifier);`
`                return Extension.ToLower() == other?.Extension.ToLower();`

File: lib/AuroraLip/Texture/Formats/GSR.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
+﻿using AuroraLib.Common;
+using AuroraLib.Common.Struct;
+using SevenZipExtractor;
```
Comment (Venomalia) [inline lib/AuroraLip/Texture/Formats/GSR.cs:3]: using SevenZipExtractor;
🥷?

File: lib/AuroraLip/Common/FormatInfo.cs (hunk 1 above)
Comment (Venomalia) [reply lib/AuroraLip/Common/FormatInfo.cs:207]: ok i see the mistake i forgot the null check at class `Identifier` && `Identifier64`
        `public bool Equals(IIdentifier other) => other.AsSpan().SequenceEqual(AsSpan());`
to 
        `public bool Equals(IIdentifier other) => other != null && other.AsSpan().SequenceEqual(AsSpan());`

my fault

Comment (iwubcode) [reply lib/AuroraLip/Common/FormatInfo.cs:207]: That works too.  Would you prefer that?

Comment (Venomalia) [reply lib/AuroraLip/Common/FormatInfo.cs:207]: yes because it fixes the actual error

Comment (iwubcode) [reply lib/AuroraLip/Common/FormatInfo.cs:207]: This does too though.. but I can fix it there instead.

Comment (iwubcode) [reply lib/AuroraLip/Common/FormatInfo.cs:207]: Ah, I see what you're saying.  It's more "direct".  Yep!

**Comment by iwubcode** [comment]
@Venomalia - I tested mips, seemed to work ok

**Pull request merged**